testutil/testdevice_linux.go

```diff
@@ -0,0 +1,97 @@
+// +build linux
+
+package testutil
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+type testDevice struct {
+	opts       TestDeviceOpts
+	fileName   string
+	deviceName string
+}
+
+func NewTestDevice(t *testing.T, opts TestDeviceOpts) TestDevice {
```

Comment:
Please leave this in btrfs package. We'll probably move this to `containerd/btrfs` over generalizing to devices.